### PR TITLE
Quick fix for Events to use Manager#_delete and adding full Invite support

### PR DESCRIPTION
**Pull request by DjDeveloperr**
This PR introduces following changes,
- \*\_DELETE events now use Manager#_delete to avoid making REST API calls through some Managers (not any atm, but in future)
- Full Invite support added - create, delete, fetch Invites.
- Added a lot of more missing REST Methods to interact with API.
- Fixed Guild#chunk
- New event: guildLoaded
- Fixed GuildChannelsManager#array
- Fixed some more things.

### CODE REVIEW

**Review by Helloyunho — APPROVED** [review]
lgtm

**Review by AkiaCode — APPROVED** [review]

File: src/structures/invite.ts [hunk 1]
```diff
@@ -30,6 +52,12 @@ export class Invite extends Base {
     this.targetUserType = data.target_user_type
     this.approximateMemberCount = data.approximate_member_count
     this.approximatePresenceCount = data.approximate_presence_count
+
+    this.uses = (data as any).uses
```
Comment (Helloyunho) [inline src/structures/invite.ts:56]: Later we should make payload for this too

**Review by Helloyunho — APPROVED** [review]
lgtm

File: src/structures/invite.ts (hunk 1 above)
Comment (DjDeveloperr) [reply src/structures/invite.ts:56]: Yea, `interface InviteMetadataPayload extends InvitePayload` 🤔

**Pull request merged**